src/test/java/de/tum/in/www1/artemis/quiz/QuizExerciseIntegrationTest.java

```diff
@@ -1322,139 +1317,93 @@ void testPerformJoin(QuizMode quizMode, ZonedDateTime release, ZonedDateTime due
+    @ValueSource(strings = { "start-now", "set-visible", "open-for-practice" })
+    void testPerformPutActionAsTutorForbidden(String action) throws Exception {
+        quizExercise = database.createAndSaveQuiz(ZonedDateTime.now().plusDays(1), null, QuizMode.SYNCHRONIZED);
 
-        request.put("/api/quiz-exercises/" + quizExercise.getId() + "/open-for-practice", quizExercise, HttpStatus.FORBIDDEN);
-        assertThat(quizExerciseRepository.findByCourseIdWithCategories(course.getId()).get(0).isIsOpenForPractice()).isFalse();
+        request.put("/api/quiz-exercises/" + quizExercise.getId() + "/" + action, quizExercise, HttpStatus.FORBIDDEN);
     }
 
     /**
-     * test non-instructors can't see the exercise if it is not set to visible
+     * test students can't see the exercise if it is not set to visible
      */
     @Test
-    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
+    @WithMockUser(username = TEST_PREFIX + "student1", roles = "USER")
     void testViewQuizExerciseAsStudentNotVisible() throws Exception {
-        final Course course = database.addCourseWithOneQuizExercise();
-        quizExercise = quizExerciseRepository.findByCourseIdWithCategories(course.getId()).get(0);
-        assertThat(quizExercise.isVisibleToStudents()).isFalse();
-        // remove instructor rights in course
-        User user = database.getUserByLogin(TEST_PREFIX + "instructor1");
-        user.setGroups(Collections.emptySet());
-        userRepo.save(user);
-        request.get("/api/quiz-exercises/" + quizExercise.getId(), HttpStatus.FORBIDDEN, QuizExercise.class);
+        quizExercise = database.createAndSaveQuiz(ZonedDateTime.now().plusDays(1), null, QuizMode.SYNCHRONIZED);
+
+        request.get("/api/quiz-exercises/" + quizExercise.getId() + "/for-student", HttpStatus.FORBIDDEN, QuizExercise.class);
     }
 
     /**
-     * test non-instructors cant delete an exercise
+     * test non-editors can't delete an exercise
      */
     @Test
-    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
+    @WithMockUser(username = TEST_PREFIX + "tutor1", roles = "TA")
     void testDeleteQuizExerciseAsNonInstructor() throws Exception {
-        final Course course = database.addCourseWithOneQuizExercise();
-        quizExercise = quizExerciseRepository.findByCourseIdWithCategories(course.getId()).get(0);
-        // remove instructor rights in course
-        User user = database.getUserByLogin(TEST_PREFIX + "instructor1");
-        user.setGroups(Collections.emptySet());
-        userRepo.save(user);
+        quizExercise = database.createAndSaveQuiz(ZonedDateTime.now().minusDays(1), null, QuizMode.SYNCHRONIZED);
+
         request.delete("/api/quiz-exercises/" + quizExercise.getId(), HttpStatus.FORBIDDEN);
     }
 
     /**
-     * test non tutors cant recalculate quiz exercise statistics
+     * test students can't recalculate quiz exercise statistics
```

Comment:
Here I would actually leave students and rename the test into testRecalculateStatisticsAsStudentForbidden since I think it is easier to read. Students is just one group of users whereas non-instructors and non-editors include 3, 2 different groups respecitvely. However, I renamed the non-editor tests into testXYZAs**NonEditor**Forbidden (before many were named testXYZAs**Tutor**Forbidden . What do you think, should I still rename students into non-tutors?

Reply:
I would still change it just to be consistent - either for everything to be `non-<role>` or for everything to be `<role below>-forbidden`. Technically [there seems to be even another role below student / user (anonymous)](https://docs.artemis.cit.tum.de/admin/accessRights), so I think it would be more fitting. I am not too confident on this though, so if you still thing the current naming is better it's ok for me if you leave it as well.

Reply:
Thanks for pointing it out, I has no idea anonymous existed. I followed your advice and renamed students to non-tutors.